app/src/main/java/sw19/moring03/paint/Views/DrawingView.java

```diff
@@ -54,32 +57,38 @@ public boolean onTouchEvent(MotionEvent event) {
                 invalidate();
                 break;
             case MotionEvent.ACTION_MOVE:
-                addPoint(xCoord, yCoord);
+                if (((MainActivity) getContext()).getChosenTool() != Tool.FILL) {
+                    addPoint(xCoord, yCoord);
+                }
                 invalidate();
                 break;
         }
         return true;
     }
 
+    @Override
+    public boolean performClick() {
```

Comment:
pretty sure you don't need to override the method if you only call the super method

Reply:
fixed with commit 7225edaa72e